pkg/api/api.go

```diff
@@ -239,7 +239,10 @@ func (a *API) RegisterRuntimeConfig(runtimeConfigHandler http.HandlerFunc) {
 func (a *API) RegisterDistributor(d *distributor.Distributor, pushConfig distributor.Config) {
 	distributorpb.RegisterDistributorServer(a.server.GRPC, d)
 
-	a.RegisterRoute("/api/v1/push", push.Handler(pushConfig.MaxRecvMsgSize, a.sourceIPs, a.cfg.SkipLabelNameValidationHeader, a.cfg.wrapDistributorPush(d)), true, false, "POST")
+	wrappedDistributor := a.cfg.wrapDistributorPush(d)
+
+	a.RegisterRoute("/api/v1/push", push.Handler(pushConfig.MaxRecvMsgSize, a.sourceIPs, a.cfg.SkipLabelNameValidationHeader, wrappedDistributor), true, false, "POST")
+	a.RegisterRoute("/api/v1/push/otlp/v1/metrics", push.OLTPHandler(pushConfig.MaxRecvMsgSize, a.sourceIPs, a.cfg.SkipLabelNameValidationHeader, wrappedDistributor), true, false, "POST")
```

Comment:
`/api/v1/push/otlp/v1/metrics` looks weird to me, with that double versioning. What if we simply use `/otlp/v1/metrics`?

Reply:
Related: https://github.com/grafana/mimir/pull/695#discussion_r811874447

Reply:
Also works. I'll open a PR with all your feedback now.